### PR TITLE
Upgrade `babashka.curl`, `babashka.fs`, and `deps-new`

### LINKED ISSUE #66: Error when invoking `clojure -M:neil`
`deps.edn`:
```clj
{:aliases
 {:neil {:deps {io.github.babashka/neil {:git/tag "v0.1.39" :git/sha "1f747a5"}}
         :main-opts ["-m" "babashka.neil"]}}}
```
```shell
$ clojure -M:neil
Syntax error (IllegalStateException) compiling fs/windows? at (babashka/neil/curl.clj:14:21).
var: #'babashka.fs/windows? is not public

Full report at:
/var/folders/tt/73f6l1j971n03fsxstmz_g280000gn/T/clojure-13803427228737455161.edn
```

### PR DESCRIPTION
**Pull request by rads**
Resolves #66.

It might be better to structure the tests so we can run _all_ the tests in both Babashka and Clojure. Since that will require more effort, I just did one test for now. The current test will at least ensure that the command is always runnable in Clojure.

### CODE REVIEW

**Comment by borkdude** [comment]
@rads Instead of testing through the generated script, we might just test through the `src/babashka/neil.clj` code without shelling out. Also I'm considering moving away from checking in the generated script since in PRs this is confusing. WDTY?

File: tests.clj [hunk 1]
```diff
@@ -154,5 +154,10 @@
         (is (= (slurp (fs/file "test-resources/new/my-scratch/deps.edn"))
                (slurp (fs/file (str target-dir "/deps.edn")))))))))
 
+(deftest clj-neil-new-test
+  (let [{:keys [out err]} (sh "clojure -M:neil new --help" {:dir "tests-clj"})]
```
Comment (borkdude) [inline tests.clj:158]: I think we could just use `babashka.deps/clojure` here. This also makes it work on Windows.

File: .github/workflows/bb-run-tests.yml [hunk 2]
```diff
@@ -6,10 +6,36 @@ jobs:
...
         uses: actions/checkout@v3
-      # First ensure we've got babashka installed
+
+      - name: Install Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 11
+
+      - name: Install Clojure
```
Comment (borkdude) [inline .github/workflows/bb-run-tests.yml:16]: Since we've already got babashka installed, there's no need to do this. See comment below.

**Comment by rads** [comment]
@borkdude:
> @rads Instead of testing through the generated script, we might just test through the src/babashka/neil.clj code without shelling out. Also I'm considering moving away from checking in the generated script since in PRs this is confusing. WDTY?

That all sounds reasonable to me. Is it OK if we take care of that in follow-up PRs?

**Comment by borkdude** [comment]
Yes

**Pull request merged**